ZanattaMichael/SRDSC
# Pull Request: CertificateThumbprint, Public Key Cert Export, and Add-SRDSCNode Update

I am submitting this pull request to address three issues related to certificate management in our codebase. Specifically, this pull request includes updates to the `CertificateThumbprint` function, adds support for exporting public key certificates to ProgramData, and updates the `Add-SRDSCNode` function to support automatic certificate enrollment.

### Changes Made
- **CertificateThumbprint:** The `CertificateThumbprint` function has been updated to include additional error handling and logging capabilities. This will help ensure that any errors related to certificate management are properly identified and addressed.
- **Public Key Cert Export:** A new function has been added to export public key certificates to ProgramData. This will allow other applications on the system to access these certificates as needed.
- **Add-SRDSCNode Update:** The `Add-SRDSCNode` function has been updated to automatically enroll nodes with the appropriate certificates. This will help streamline the process of adding new nodes to the system and ensure that they are properly authenticated.

### How to Test
To test these changes, please run the following commands:
1. Run the updated `CertificateThumbprint` function with various inputs to ensure that error handling and logging work as expected.
2. Use the new function to export a public key certificate to ProgramData and verify that it can be accessed by other applications on the system.
3. Use the updated `Add-SRDSCNode` function to add a new node to the system and verify that it is properly enrolled with the appropriate certificates.

### Additional Notes
These changes were made in response to feedback from users who have experienced issues related to certificate management in our codebase. We believe that these updates will help improve the overall functionality and reliability of our system.
